### PR TITLE
Config support for Telemetry service

### PR DESCRIPTION
**Pull request by tamilmani1989**
…/networkmonitor

2. Added telemetry config through which we can configure reportTohost interval
3. Added unit tests for changes

**What this PR does / why we need it**:
1. This PR adds support for telemetry service to read config file which provides a way to configure time interval for sending telemetry payload to host
2. Moved telemetry specific functions to telemetry package so that other components can use
3. Start telemetry process with argument which indicates config directory ( eg: azure-vnet-telemetry -d <dir_path>). By default it will look for config in /opt/cni/bin.
4. Moved log initialization in telemetry service process code. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Run unit tests for linux and windows. Did sanity testing in linux and windows. For multitenancy, requires changes in cns/dnc which will be a separate PR

**Release note**:

```release-note
```

### CODE REVIEW

File: telemetry/metadata_test.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"location":"eastus","name":"k8s-agentpool1-42685608-0","offer":"aks","osType":"Linux","placementGroupId":"","platformFaultDomain":"0","platformUpdateDomain":"0","publisher":"microsoft-aks","resourceGroupName":"rgcnideftesttamil","sku":"aks-ubuntu-1604-201902","subscriptionId":"9b8218f9-902a-4d20-a65c-e98acec5362f","tags":"aksEngineVersion:canary;creationSource:aksengine-k8s-agentpool1-42685608-0;orchestrator:Kubernetes:1.10.13;poolName:agentpool1;resourceNameSuffix:42685608","version":"2019.02.12","vmId":"6baf785b-397c-4967-9f75-cdb3d0df66c4","vmSize":"Standard_DS2_v2","KernelVersion":""}
```
Comment (jaer-tsun) [inline telemetry/metadata_test.json]: Is that one of our subscription ID?

Comment (tamilmani1989) [reply telemetry/metadata_test.json]: sure

File: telemetry/telemetrybuffer.go [hunk 2]
```diff
@@ -426,3 +420,47 @@ func getHostMetadata() (Metadata, error) {
...
+}
+
+// ReadConfigFile - Read telemetry config file and populate to structure
+func ReadConfigFile(filePath string) (TelemetryConfig, error) {
+	config := TelemetryConfig{}
+
+	b, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Printf("[Telemetry] Failed to read telemetry config: %v", err)
+		return config, err
```
Comment (jaer-tsun) [inline telemetry/telemetrybuffer.go]: we should return nil config here, no need for empty

File: telemetry/telemetrybuffer.go [hunk 3]
```diff
@@ -426,3 +420,47 @@ func getHostMetadata() (Metadata, error) {
...
+	}
+
+	if err = json.Unmarshal(b, &config); err != nil {
+		log.Printf("[Telemetry] unmarshal failed with %v", err)
```
Comment (jaer-tsun) [inline telemetry/telemetrybuffer.go]: same here, let's return nil config instead of empty

File: telemetry/telemetrybuffer.go (hunk 2 above)
Comment (tamilmani1989) [reply telemetry/telemetrybuffer.go]: nope..this is good instead of nil..return by value is better than returning pointers in golang

Comment (jaer-tsun) [reply telemetry/telemetrybuffer.go]: I see that the config is really small so it's ok

**Pull request merged**